### PR TITLE
Backend init

### PR DESCRIPTION
**Pull request by avi90**
Build typescript stuff with `npm run build:tsc`. This builds the whole project and resolves the dependencies between all projects in the monorepo.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-node_modules
\ No newline at end of file
+node_modules/
```
Comment (p4dd9) [inline .gitignore:1]: nit: '/' not needed here

File: applications/backend/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+    "root": true,
+    "parser": "@typescript-eslint/parser",
+    "plugins": [
+      "@typescript-eslint"
+    ],
+    "extends": [
+      // "eslint:recommended",
```
Comment (p4dd9) [inline applications/backend/.eslintrc:8]: Do we need this in the future for the backend application? If not, please delete

File: package.json [hunk 3]
```diff
@@ -10,7 +10,9 @@
 		"url": "git+https://github.com/Ludecat/catcrowdparty.git"
 	},
 	"scripts": {
-		"build:all": "npm run build --workspaces"
+		"build:all": "npm run build --workspaces",
+		"build:tsc": "tsc --build",
```
Comment (p4dd9) [inline package.json:14]: Let's move typescript compiler scripts to backend project?

File: applications/backend/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+		logger.info(`destroying..`)
+		socket.emit(DESTROY_CHECKBOXES)
+	}, 3000)
+
+	socket.on('disconnect', (reason) => {
+		logger.info(`socket ${socket.id} disconnected with reason: ${reason}`)
+	})
+})
+
+httpServer.listen(5000)
```
Comment (p4dd9) [inline applications/backend/src/index.ts:23]: nit: export port to `shared.ts` and re-use in frontend application

File: package.json (hunk 3 above)
Comment (p4dd9) [reply package.json:14]: Ah, this will also build the `common` package

File: applications/backend/src/index.ts (hunk 4 above)
Comment (avi90) [reply applications/backend/src/index.ts:23]: Don't think shared is a good place either. Should live in the environment somewhere.

**Pull request merged**